xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentfield.vm

```diff
@@ -0,0 +1,59 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+###
+### Return a single comment field
+###
+###
+## requested field name 
+#set ($name = $request.name)
+## the number of the comment, if no number is provided we generate a field for a new comment.
+#set ($number = $numbertool.toNumber($request.number).intValue())
+
+## the 'comment' field from the XWikiComments class is used as a template for the field to display
+#set ($xCommentClass = 'XWiki.XWikiComments')
+#set ($commentClass = $xwiki.getDocument($xCommentClass).getxWikiClass())
+
+## the number of the comment, if no number is provided we generate a field for a new comment.
+#if ($number && $number > -1)
+  #set ($commentObj = $doc.getObject($xCommentClass, $number))
+  #set ($value = $commentObj.getValue('comment'))
+#else
+  #set ($commentObj = $doc.newObject($xCommentClass))
+  #set ($value = '')
+#end
+
+## We need to know the type of editor used for comments.
```

Comment:
AFAICS it's used for the comments.js at least to enable/disable preview button depending on the editor. Now as I specified on Matrix, I'm not even entirely sure if we should disable preview button when using the WYSIWYG editor... wdyt?

Reply:
> it's used for the comments.js at least to enable/disable preview button depending on the editor.

I figured it out, but the comment should mention **why** it's needed, not only that it's needed :)

> I'm not even entirely sure if we should disable preview button when using the WYSIWYG editor... wdyt?

The preview button doesn't bring any value to the WYSIWYG editor so I would keep it hidden in order to have a cleaner / simpler UI. I don't remember deciding to keep the preview button in the stand-alone WYSIWYG mode. We didn't remove it there just because we were lazy I think.